services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/application.py

```diff
@@ -171,8 +177,8 @@ def create_app() -> FastAPI:
     app.state.application_health = ApplicationHealth()
     application_settings: ApplicationSettings = app.state.settings
 
-    if application_settings.DYNAMIC_SIDECAR_TRACING:
-        setup_tracing(app, application_settings.DYNAMIC_SIDECAR_TRACING, PROJECT_NAME)
+    if get_tracing_data(app).tracing_enabled:
```

Comment:
since you are using get_tracing_data(app) twice in this function, maybe don't call it twice and assign it to a variable